### PR TITLE
Handle riker stopped cmds

### PR DESCRIPTION
**Pull request by gliargovas**
Modification to handle killed Riker containers that tried to perform network operations.

This addresses @ericzty 's modification of Riker.

### CODE REVIEW

**Comment by angelhof** [comment]
Do we have a test to check that this works? I think it is a good idea to add one when adding new code.

Also, it is high time to add the tests in CI. This is a good pointer to start: https://github.com/binpash/pash/blob/main/.github/workflows/tests.yaml#L25.

File: parallel-orch/partial_program_order.py [hunk 1]
```diff
@@ -380,15 +398,22 @@ def speculate_cmd_non_blocking(self, node_id: int):
...
+    def command_execution_completed(self, node_id: int, exit_code:int, sandbox_dir: str):
         self.sandbox_dirs[node_id] = sandbox_dir
         _proc, trace_file = self.commands_currently_executing.pop(node_id)
+        # Handle stopped by riker due to network access
+        if int(exit_code) == 159:
+            logging.debug(f" Adding {node_id} to stopped")
+            self.stopped.add(node_id)
+        else:
+            logging.debug(f" Removing {node_id} from stopped")
+            self.stopped.discard(node_id)
```
Comment (angelhof) [inline parallel-orch/partial_program_order.py]: Why is this needed?

**Review by angelhof — COMMENTED** [review]
Great job with this PR! I am OK with merging this once we have a test that we know passes and covers this code at east partially. In a future PR we should make the CI though!

**Comment by gliargovas** [comment]
For now, I just tested it by returning a non-zero ec in every sandbox execution. @ericzty is working on Riker in order to add a --frontier flag, which will suppress Riker's checks for forbidden syscalls. We will call Riker like this outside of the sandbox.

**Comment by gliargovas** [comment]
I will add a test that might probably fail for now (until we fix the above).

File: parallel-orch/partial_program_order.py (hunk 1 above)
Comment (gliargovas) [reply parallel-orch/partial_program_order.py]: Frontier cmds might not exist inside the workset anymore, because they might be stopped. We invoke these cmds in `--frontier` mode (outside of sandbox) so next time they execute.

Makes more sense to attempt to remove them from stopped (if they are stopped) when starting the re-execution of the frontier. Moved the removal there.

**Comment by angelhof** [comment]
It is OK if the test fails for now, we can comment it out, but it is still good to have it.

**Comment by angelhof** [comment]
Feel free to merge when you add the test :)
